maven-plugin-api/src/main/java/org/apache/maven/plugin/descriptor/MojoDescriptor.java

```diff
@@ -85,8 +84,8 @@ public class MojoDescriptor extends ComponentDescriptor<Mojo> implements Cloneab
     private String executeLifecycle;
 
     /**
-     * Description with reason of Mojo deprecation. Similar to Javadoc {@code @deprecated}.
-     * This will trigger a warning when a user tries to use a Mojo marked as deprecated.
+     * Specify the version when the Mojo was deprecated to the API. Similar to Javadoc deprecated. This will trigger a
```

Comment:
Although mentioning the version when this was deprecated is nice to have, the must have info is what to use instead.

Reply:
In master it looks different: https://github.com/apache/maven/blob/23d13fb322b8d66f03f02ece57d75e154f847367/maven-plugin-api/src/main/java/org/apache/maven/plugin/descriptor/MojoDescriptor.java#L88. Seems that for some reason this PR restores the 12 year old revision https://github.com/apache/maven/commit/acc04dca1b6a7ef39f41d976dcc8eb96aa0aa53f.

Reply:
Let me check...

Reply:
The only changes to this file on master I can find are 47e2d689d687358a85c2cdecc176cac36093817e and 5ce5591ae341188230d6ecca76c90bd0a786fcb2

I agree that they should be included as well.

Reply:
@kwin Now the diff is only v4 API related. Please check again.